Documentation/tutorial/intro_template.md

```diff
@@ -0,0 +1,170 @@
+Introduction to *DocFX Template System*
+===============================
+
+*DocFX Template System* provides a flexible way of defining and using templates.
+
+As the following DocFX workflow shows,
+
+![DocFX workflow](../spec/images/docfx_workflow_highlevel.png)
+
+DocFX loads the set of files and transforms them into different data models using different type of document processors. Afterwards, *DocFX Template System* loads these data models, and transforms them into output files based on the document type of the data model.
```

Comment:
Seems you're using "document" as the term of the basic unit in template. Add a definition for document and give some examples.